plugins/inputs/glusterfs/glusterfs.go

```diff
@@ -0,0 +1,71 @@
+package glusterfs
+
+// glusterfs.go
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	"bufio"
+	"os/exec"
+	"regexp"
+	"strconv"
+)
+
+var matchBrick = regexp.MustCompile("^Brick: (.*)$")
+var matchRead = regexp.MustCompile("Data Read: ([0-9]+) bytes$")
+var matchWrite = regexp.MustCompile("Data Written: ([0-9]+) bytes$")
+
+type GlusterFS struct {
+	Volumes []string
+}
+
+func (gfs *GlusterFS) Description() string {
+	return "Plugin reading values from the GlusterFS profiler"
+}
+
+func (gfs *GlusterFS) SampleConfig() string {
+	return "volumes = [\"volume-name\"]"
+}
+
+func (gfs *GlusterFS) Gather(acc telegraf.Accumulator) error {
+	for _, volume := range gfs.Volumes {
+		var cmdName = "sudo"
+		var cmdArgs = []string{"gluster", "volume", "profile", volume, "info", "cumulative"}
+
+		cmd := exec.Command(cmdName, cmdArgs...)
+		cmdReader, err := cmd.StdoutPipe()
+		if err != nil {
+			continue
+		}
+
+		scanner := bufio.NewScanner(cmdReader)
+		go func() {
```

Comment:
As mentioned, I don't know how to Go :). You are saying it'd work just the same if I removed the go func and just executed that code there ? I'll give it a try tomorrow then, thanks.